src/lib/uni/src/tags/tag.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * A bundle used for comparing conflicting CSS styles
+ * generated by multiple tags.
+ */
+export interface ValuePriority {
+  /** The value of a corresponding CSS style */
+  value: string;
+  /**
+   * The priority of the associated value compared to
+   * other comparable values
+   */
+  priority: number;
+}
+
+/**
+ * A tag is a way of consolidating logic about a style pattern. For instance,
+ * setting the 'flex' attribute could be done with a Flex Tag. Each tag has an
+ * associated name, builder, cache, and dependencies on other builder input
+ * values.
+ */
+export abstract class Tag {
```

Comment:
In this implementation, `Tag`s are attributes? I think that this can be easily confused with [HTML tags](https://developer.mozilla.org/en-US/docs/Glossary/Tag).